src/DocChecks.jl

```diff
@@ -226,13 +226,17 @@ end
 
 # Regex used here to replace gensym'd module names could probably use improvements.
 function checkresult(sandbox::Module, result::Result, doc::Documents.Document)
-    mod_regex = Regex("(Symbol\\(\"$(sandbox)\"\\)|$(sandbox))[,.]")
+    sandbox_name = module_name(sandbox)
+    mod_regex = Regex("(Main\\.)?(Symbol\\(\"$(sandbox_name)\"\\)|$(sandbox_name))[,.]")
+    mod_regex_nodot = Regex(("(Main\\.)?$(sandbox_name)"))
```

Comment:
Is "Main" optional in these regexes because of 0.6 / 0.7 differences? If so, we could maybe have a small note here saying we should clean it up once 0.6 support is dropped.

Reply:
The difference is in e.g.

```
julia> module M
       struct P end
       @show P
       end;
```

**0.6**

```
P = M.P
```

**0.7**

```
P = Main.M.P
```

However, I was not sure if there are still places where `M.P` is printed so I left it optional.

Reply:
Hmm.. and I guess JuliaLang/julia#23806 would revert back to just `M.P`, right?

Reply:
No, I don't think so since that is only for types and the type `P` is not reachable from `Main`.

Reply:
Yes, for example:

```
julia> module M
           sqrt(3)
           sqrt = 2
       end
ERROR: cannot assign variable Base.sqrt from module M
```

Reply:
Right, never mind this comment then :smile: